node/peer.js

```diff
@@ -95,6 +89,24 @@ function TChannelPeer(channel, hostPort, options) {
 
 inherits(TChannelPeer, EventEmitter);
 
+TChannelPeer.prototype.setPreferConnectionDirection = function setPreferConnectionDirection(direction) {
+    var self = this;
+    if (self.preferConnectionDirection === direction) {
+        return;
+    }
```

Comment:
Given that @jcorbin has suggested that each peer should only have on strategy since that is the norm in current practice, perhaps we should assert that all subchannels using that subchannel agree on a single strategy, or warn if they conflict. We should definitely not allow subchannels to thrash on strategy.